.github/workflows/build_wheels.yml

```diff
@@ -111,8 +111,8 @@ jobs:
           password: ${{ secrets.PYPI_API_TOKEN }}
 
       - name: Upload to TestPyPI
-        # otherwise, upload to TestPyPi
-        if: ${{ !startsWith(github.ref, 'refs/tags/v') }}
+        # otherwise, upload to TestPyPi when secrets are available
+        if: ${{ !startsWith(github.ref, 'refs/tags/v') && (github.event_name != 'pull_request' || (github.event.pull_request.head.repo.full_name == github.repository && github.actor != 'dependabot[bot]')) }}
```

Comment:
Should we also exclude `pre-commit-ci`? https://github.com/glotzerlab/freud/pull/765

Reply:
Those PRs are passing, so it seems we don't need to now. We could add the exception if we think `pre-commit` pull requests might run with reduced privileges in the future. However, the pre-commit auto-fix commits are applied directly to user-generated pull requests, so the app already has indirect access to our secrets.